includes/abstracts/abstract-wc-order.php

```diff
@@ -969,7 +969,7 @@ public function apply_coupon( $raw_coupon ) {
 		if ( 0 === $this->get_customer_id() ) {
 			$data_store  = $coupon->get_data_store();
 			$usage_count = $data_store->get_usage_by_email( $coupon, $this->get_billing_email() );
-			if ( $usage_count >= $coupon->get_usage_limit_per_user() ) {
+			if ( ! empty( $usage_count ) && $usage_count >= $coupon->get_usage_limit_per_user() ) {
```

Comment:
I just tested, checking for `! empty( $usage_count )` doesn't fix this problem, should works only for the first time for each email address, after will start showing the same message again, because we need to check for the setting and not how many times some user have used that coupon.
So in order to properly fix it, need to check if `$coupon->get_usage_limit_per_user()` is greater than `0`, see:

```php
if ( 0 < $coupon->get_usage_limit_per_user() && $usage_count >= $coupon->get_usage_limit_per_user() ) {
```

Reply:
Yeah. Make sense.

Reply:
Check it now and let me know if you need any help for this.